src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaCapturedVariable.cs

```diff
@@ -65,6 +65,11 @@ private static string GetCapturedVariableFieldName(Symbol variable, ref int uniq
                 {
                     return GeneratedNames.MakeHoistedLocalFieldName(local.SynthesizedKind, uniqueId++);
                 }
+
+                if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+                {
+                    return GeneratedNames.MakeSynthesizedInstrumentationPayloadLocalName(uniqueId++);
+                }
```

Comment:
If I understand correctly, now the payload is in a local inside the instrumented method and this means that lambdas can capture it. Do you have a test with lambdas getting instrumented?